src/librustc_mir/transform/simplify_try.rs

```diff
@@ -397,7 +442,9 @@ impl LocalUseCounter {
 
 impl<'tcx> Visitor<'tcx> for LocalUseCounter {
     fn visit_local(&mut self, local: &Local, context: PlaceContext, _location: Location) {
-        if context.is_storage_marker() {
+        if context.is_storage_marker()
+            || context == PlaceContext::NonUse(NonUseContext::VarDebugInfo)
```

Comment:
are there any `NonUseContext` that are relevant here? The only one left that isn't covered is `AscribeUserTy`. So this could just be `if let `PlaceContext::NonUse(_) = context` I think.

Reply:
Hmm I guess those are unused after NLL runs right? Maybe we should have a separate pass that runs at the beginning of the optimization phase and removes all of those?

Reply:
true, and then changing this site to the general match on `NonUseContext` doesn't actually change anything, anymore, so let's do that in a separate PR